crates/hir_def/src/type_ref.rs

```diff
@@ -163,14 +184,15 @@ impl TypeRef {
                     types.iter().for_each(|t| go(t, f))
                 }
                 TypeRef::RawPtr(type_ref, _)
-                | TypeRef::Reference(type_ref, _)
+                | TypeRef::Reference(type_ref, ..)
                 | TypeRef::Array(type_ref)
                 | TypeRef::Slice(type_ref) => go(&type_ref, f),
                 TypeRef::ImplTrait(bounds) | TypeRef::DynTrait(bounds) => {
                     for bound in bounds {
                         match bound {
                             TypeBound::Path(path) => go_path(path, f),
-                            TypeBound::Error => (),
+                            TypeBound::ForLifetime(..) => (), // FIXME: ForLifetime
```

Comment:
that's a regression, I think we should keep treating `for<> foo` like `foo` while we don't fully support it yet

Reply:
... or maybe just not introduce `ForLifetime` right now

Reply:
That TypeBound variant is actually never constructed so it shouldn't be a regression. I can remove this nevertheless if you want, since it's not used currently anyways.